crates/uv-python/src/sysconfig/parser.rs

```diff
@@ -0,0 +1,394 @@
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize)]
+#[serde(untagged)]
+pub(super) enum Value {
+    String(String),
+    Int(i32),
+}
+
+/// The data extracted from a `_sysconfigdata_` file.
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize)]
+pub(super) struct SysconfigData(BTreeMap<String, Value>);
+
+impl SysconfigData {
+    /// Returns an iterator over the key-value pairs in the map.
+    pub(super) fn iter_mut(&mut self) -> std::collections::btree_map::IterMut<String, Value> {
+        self.0.iter_mut()
+    }
+
+    /// Inserts a key-value pair into the map.
+    pub(super) fn insert(&mut self, key: String, value: Value) -> Option<Value> {
+        self.0.insert(key, value)
+    }
+
+    /// Formats the `sysconfig` data as a pretty-printed string.
+    pub(super) fn to_string_pretty(&self) -> Result<String, serde_json::Error> {
+        let output = {
+            let mut buf = Vec::new();
+            let mut serializer = serde_json::Serializer::with_formatter(
+                &mut buf,
+                PrettyFormatter::with_indent(b"    "),
+            );
+            self.0.serialize(&mut serializer)?;
+            String::from_utf8(buf).unwrap()
+        };
+        Ok(format!(
+            "# system configuration generated and used by the sysconfig module\nbuild_time_vars = {output}\n",
+        ))
+    }
+}
+
+impl std::fmt::Display for SysconfigData {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let output = {
+            let mut buf = Vec::new();
+            let mut serializer = serde_json::Serializer::new(&mut buf);
+            self.0.serialize(&mut serializer).unwrap();
+            String::from_utf8(buf).unwrap()
+        };
+        write!(f, "{output}",)
+    }
+}
+
+impl FromIterator<(String, Value)> for SysconfigData {
+    fn from_iter<T: IntoIterator<Item = (String, Value)>>(iter: T) -> Self {
+        Self(iter.into_iter().collect())
+    }
+}
+
+/// Parse the `_sysconfigdata_` file (e.g., `{real_prefix}/lib/python3.12/_sysconfigdata__darwin_darwin.py"`
+/// on macOS).
+///
+/// `_sysconfigdata_` is structured as follows:
+///
+/// 1. A comment on the first line (e.g., `# system configuration generated and used by the sysconfig module`).
+/// 2. An assignment to `build_time_vars` (e.g., `build_time_vars = { ... }`).
+///
+/// The right-hand side of the assignment is a JSON object. The keys are strings, and the values
+/// are strings or numbers.
+impl FromStr for SysconfigData {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Read the first line of the file.
+        let Some(s) =
+            s.strip_prefix("# system configuration generated and used by the sysconfig module\n")
+        else {
+            return Err(Error::MissingHeader);
+        };
+
+        // Read the assignment to `build_time_vars`.
+        let Some(s) = s.strip_prefix("build_time_vars = ") else {
```

Comment:
Is there a chance that the whitespace around `=` might be missing? I think the data is auto-generated so it's quite unlikely.